src/OpenTK/Platform/X11/X11GLNative.cs

```diff
@@ -1681,44 +1682,46 @@ public override bool CursorVisible
             get { return cursor_visible; }
             set
             {
-                if (value && !cursor_visible)
+                using (new XLock(window.Display))
                 {
-                    using (new XLock(window.Display))
-                    {
-                        UngrabMouse();
-
-                        Point p = PointToScreen(new Point(MouseState.X, MouseState.Y));
-                        Mouse.SetPosition(p.X, p.Y);
-
-                        // Note: if cursorHandle = IntPtr.Zero, this restores the default cursor
-                        // (equivalent to calling XUndefineCursor)
-                        Functions.XDefineCursor(window.Display, window.Handle, cursorHandle);
-                        cursor_visible = true;
-                    }
+                    HideCursor(!value);
+                    cursor_visible = value;
                 }
-                else if(!value && cursor_visible)
+            }
+        }
+
+        public override bool CursorGrabbed
+        {
+            get { return cursor_grabbed; }
+            set
+            {
+                using (new XLock(window.Display))
                 {
-                    using (new XLock(window.Display))
-                    {
-                        GrabMouse();
-                        cursor_visible = false;
-                    }
+                    GrabCursor(value);
+                    cursor_grabbed = value;
                 }
             }
         }
 
-        private void GrabMouse()
+        private void GrabCursor(bool grab)
         {
-            Functions.XGrabPointer(window.Display, window.Handle, false,
-                EventMask.PointerMotionMask | EventMask.ButtonPressMask |
-                EventMask.ButtonReleaseMask,
-                GrabMode.GrabModeAsync, GrabMode.GrabModeAsync,
-                window.Handle, EmptyCursor, IntPtr.Zero);
+            if (grab)
+            {
+                Functions.XGrabPointer(window.Display, window.Handle, false,
+                                       EventMask.PointerMotionMask | EventMask.ButtonPressMask |
+                                       EventMask.ButtonReleaseMask | EventMask.FocusChangeMask,
+                                       GrabMode.GrabModeAsync, GrabMode.GrabModeAsync,
+                                       window.Handle, IntPtr.Zero, IntPtr.Zero);
+            }
+            else
+            {
+                Functions.XUngrabPointer(window.Display, IntPtr.Zero);
+            }
         }
 
-        private void UngrabMouse()
+        private void HideCursor(bool hide)
```

Comment:
What's up with this? HideCursor with a boolean? If it can both hide and unhide the mouse, split it into two methods without a parameter or rename the method.